core/state_processor_rip7560.go

```diff
@@ -108,25 +128,23 @@ func handleRip7560Transactions(transactions []*types.Transaction, index int, sta
 		}
 		statedb.IntermediateRoot(true)
 
-		vpr.Payment = payment
-		vpr.PrepaidGas = prepaidGas
 		validationPhaseResults = append(validationPhaseResults, vpr)
 		validatedTransactions = append(validatedTransactions, tx)
 	}
 
 	// This is the line separating the Validation and Execution phases
-	// It should be separated to implement the mempool-friendly AA RIP (number not assigned yet)
 	for i, vpr := range validationPhaseResults {
 		// TODO: this will miss all validation phase events - pass in 'vpr'
 		statedb.SetTxContext(vpr.Tx.Hash(), i)
 		log.Info("[RIP-7560] Execution Phase", "i", i)
-		executionResult, paymasterPostOpResult, cumulativeGasUsed, err := ApplyRip7560ExecutionPhase(chainConfig, vpr, bc, coinbase, gp, statedb, header, cfg, vpr.Payment, vpr.PrepaidGas)
+		executionResult, paymasterPostOpResult, gasUsed, err := ApplyRip7560ExecutionPhase(chainConfig, vpr, bc, coinbase, gp, statedb, header, cfg)
 
-		root := statedb.IntermediateRoot(true).Bytes()
-		receipt := &types.Receipt{Type: vpr.Tx.Type(), PostState: root, CumulativeGasUsed: cumulativeGasUsed}
+		receipt := &types.Receipt{Type: vpr.Tx.Type(), TxHash: vpr.Tx.Hash(), GasUsed: gasUsed, CumulativeGasUsed: gasUsed}
```

Comment:
Is there any reason why you're excluding the PostState information from the receipt?